src/Optimization/hiopKKTLinSysMDS.cpp

```diff
@@ -120,245 +156,148 @@ namespace hiop
+    Msys.setToZero();
+  
+    int alpha = 1.;
+
+    // perf eval 
+    //hiopTimer tm;
+    //tm.start();
 
-	HessMDS_->de_mat()->addUpperTriangleToSymDenseMatrixUpperTriangle(0, alpha, Msys);
-	Jac_cMDS_->de_mat()->transAddToSymDenseMatrixUpperTriangle(0, nxd,     alpha, Msys);
-	Jac_dMDS_->de_mat()->transAddToSymDenseMatrixUpperTriangle(0, nxd+neq, alpha, Msys);
+    HessMDS_->de_mat()->addUpperTriangleToSymDenseMatrixUpperTriangle(0, alpha, Msys);
+    Jac_cMDS_->de_mat()->transAddToSymDenseMatrixUpperTriangle(0, nxd,     alpha, Msys);
+    Jac_dMDS_->de_mat()->transAddToSymDenseMatrixUpperTriangle(0, nxd+neq, alpha, Msys);
 
-	//tm.stop();
-	//printf("the three add methods took %g sec\n", tm.getElapsedTime());
-	//tm.reset();
+    //tm.stop();
+    //printf("the three add methods took %g sec\n", tm.getElapsedTime());
+    //tm.reset();
 
-	//update -> add Dxd to (1,1) block of KKT matrix (Hd = HessMDS_->de_mat already added above)
-	Msys.addSubDiagonal(0, alpha, *Dx_, nxs, nxd);
-	//add perturbation 'delta_wx' for xd
-	Msys.addSubDiagonal(0, nxd, delta_wx);
+    //update -> add Dxd to (1,1) block of KKT matrix (Hd = HessMDS_->de_mat already added above)
+    Msys.addSubDiagonal(0, alpha, *Dx_, nxs, nxd);
+    //add perturbation 'delta_wx' for xd
+    Msys.addSubDiagonal(0, nxd, delta_wx);
 	
-	//build the diagonal Hxs = Hsparse+Dxs
-	if(NULL == Hxs_) {
-	  Hxs_ = LinearAlgebraFactory::createVector(nxs); assert(Hxs_);
-	}
-	Hxs_->startingAtCopyFromStartingAt(0, *Dx_, 0);
+    //build the diagonal Hxs = Hsparse+Dxs
+    if(NULL == Hxs_) {
+      Hxs_ = LinearAlgebraFactory::createVector(nxs); assert(Hxs_);
+    }
+    Hxs_->startingAtCopyFromStartingAt(0, *Dx_, 0);
 	
-	//a good time to add the IC 'delta_wx' perturbation
-	Hxs_->addConstant(delta_wx);
-
+    //a good time to add the IC 'delta_wx' perturbation
+    Hxs_->addConstant(delta_wx);
 
-	//Hxs +=  diag(HessMDS->sp_mat());
-	//todo: make sure we check that the HessMDS->sp_mat() is a diagonal
-	HessMDS_->sp_mat()->startingAtAddSubDiagonalToStartingAt(0, alpha, *Hxs_, 0);
-	nlp_->log->write("Hxs in KKT_MDS_X", *Hxs_, hovMatrices);
+    //Hxs +=  diag(HessMDS->sp_mat());
+    //todo: make sure we check that the HessMDS->sp_mat() is a diagonal
+    HessMDS_->sp_mat()->startingAtAddSubDiagonalToStartingAt(0, alpha, *Hxs_, 0);
+    nlp_->log->write("Hxs in KKT_MDS_X", *Hxs_, hovMatrices);
 
+    //add - Jac_c_sp * (Hxs)^{-1} Jac_c_sp^T to diagonal block linSys starting at (nxd, nxd)
+    alpha = -1.;
 
-	//add - Jac_c_sp * (Hxs)^{-1} Jac_c_sp^T to diagonal block linSys starting at (nxd, nxd)
-	alpha = -1.;
+    // perf eval
+    //tm.start();
+    Jac_cMDS_->sp_mat()->addMDinvMtransToDiagBlockOfSymDeMatUTri(nxd, alpha, *Hxs_, Msys);
 
-	// perf eval
-	//tm.start();
-	Jac_cMDS_->sp_mat()->addMDinvMtransToDiagBlockOfSymDeMatUTri(nxd, alpha, *Hxs_, Msys);
+    //tm.stop();
+    //printf("addMDinvMtransToDiagBlockOfSymDeMatUTri 111 took %g sec\n", tm.getElapsedTime());
+    //tm.reset();
```

Comment:
Seems to be quite a bit of timing code left in here

Reply:
@cnpetra what's your opinion here? I noticed that you have some time-tracking functions commented out in the code. Should we remove them, or move them into DEEPCHECK mode?

Reply:
leave them commented, I use those often to time things...

Reply:
@ashermancinelli all the points you mentioned have been addressed. :-)